specification/cognitiveservices/data-plane/AzureOpenAI/authoring/preview/2023-04-15-preview/images.json

```diff
@@ -0,0 +1,527 @@
+      "required": true,
+      "type": "string",
+      "format": "url",
+      "x-ms-skip-url-encoding": true,
+      "x-ms-parameter-location": "client"
+    },
+    "apiVersionQueryParameter": {
+      "in": "query",
+      "name": "api-version",
+      "description": "The requested API version.",
+      "required": true,
+      "type": "string",
+      "x-ms-client-default": "2023-04-15-preview",
+      "x-ms-parameter-location": "client"
+    }
+  },
+  "paths": {
+    "/images/generate": {
+      "post": {
+        "summary": "Generates a batch of images from a text caption",
+        "operationId": "Images_Generate",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionQueryParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "The specification of the images that should be generated.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GenerateImagesCreation"
+            }
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Created",
+            "headers": {
+              "operation-location": {
+                "type": "string",
+                "format": "url",
+                "description": "URL to poll for the operation result."
+              },
+              "x-ms-request-id": {
+                "type": "string",
+                "format": "uuid",
+                "description": "Service generated correlation id identifying the request, in the form of a GUID."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/GenerateImagesResponse"
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "The error code indicating what went wrong."
+              },
+              "x-ms-request-id": {
+                "type": "string",
+                "format": "uuid",
+                "description": "Service generated correlation id identifying the request, in the form of a GUID."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Creating an image generation operation.": {
+            "$ref": "./examples/post_images_generate.json"
```

Comment:
>post_images_generate [](http://example.com/codeflow?start=32&length=20)

create_image.json?

Reply:
I think like this it is good, since we are basically creating images in all three APIs (generate, edit and vary). Having post_images_generate, post_images_edit etc. makes the names more precise.